recovery/elliptics_recovery/iterator.py

```diff
@@ -379,6 +386,24 @@ def iterate_with_stats(cls, node, eid, timestamp_range,
         return result, result_len
 
 
+class MergeRecoveryIterator(Iterator):
+    '''
+    This class is used in merge recovery for backend iteratation on ranges which are not belong to
+    it using copy iterator. Every iterated key is moved to the backend, where it should exists.
+    If moving of some key was failed, then it saves the key to the results container.
+    '''
+    def __init__(self, *args,  **kwargs):
+        super(MergeRecoveryIterator, self).__init__(*args, **kwargs)
+
+    def _start_iterator(self, eid, ranges, flags, timestamp_range):
+        flags |= elliptics.iterator_flags.move
+        return self.session.start_copy_iterator(eid, ranges, [eid.group_id], flags, timestamp_range[0], timestamp_range[1])
+
+    def _on_key_response(self, results, record):
+        if record.response.status != 0:
```

Comment:
It should also update statistics in own way, because now it will not show any recovering progress and all iterator's statistics will be about keys which weren't recovered by copy-iterator.